docs/taxonomy_mapping.md

```diff
@@ -0,0 +1,51 @@
+
+This means that not all of the CIPD's dimensions and subdimensions of job quality are in our taxonomy, because some do not appear or cannot be inferred from job adverts. For example, trade union existence and activity is not something that is typically mentioned in job adverts. Similarly, the subdimension "use of skills" refers to whether an employee is in employment that makes use of their specific skillset, so it is dependent on the individual and cannot be inferred from the job advert.
+
+You can find the final taxonomy [here](https://open-jobs-lake.s3.eu-west-1.amazonaws.com/job_quality/keywords/keyword_lookup+-+v8.csv).
+
+## Using a different taxonomy
+
+We developed a taxonomy that fit our purposes, but it is possible to use your own taxonomy - though we cannot guarantee performance, as our pipeline has been evaluated against the specific taxonomy we developed. If you wish to use your own taxonomy, the steps that you will need to follow are:
+
+1. Modify the getter function [get_keywords()](https://github.com/nestauk/dap_job_quality/blob/dev/dap_job_quality/getters/keywords.py) so that it points to your taxonomy file. This should be a dataframe (saved as `.csv`, `.parquet` or similar) with the columns `dimension`, `sub_category`, `target_phrase`. `target_phrase` contains the strings that will be embedded and compared to the job advert text.
+2. You should evaluate the performance of the pipeline against your taxonomy. This can be done using the scripts in [`dap_job_quality/analysis/mapping_evaluation/`](https://github.com/nestauk/dap_job_quality/blob/dev/dap_job_quality/analysis/README.md) and the notebook [`dap_job_quality/notebooks/Evalution.ipynb`](https://github.com/nestauk/dap_job_quality/blob/dev/dap_job_quality/notebooks/Evaluation.ipynb).
+
+## Mapping approach
+
+The overall approach taken is to chunk up an input sentence into pieces that are then compared to target phrases from a taxonomy.
+
+1. **Initial cleaning**: The input sentence is chunked up into smaller pieces. At first, it is split up on characters that commonly indicate a list, eg ":" (see [`split_text()`](https://github.com/nestauk/dap_job_quality/blob/647565433a4ce21e510de0b574443d516fc5a037/dap_job_quality/pipeline/find_job_quality.py#L91)). Digits are also replaced with 'X' because for our purposes, the exact number is not important: for example, we would like our pipeline to treat "25 days of annual leave" and "30 days of annual leave" as the same.
+
+2. **Sentence chunking**: Then, if the sentence is 6 words or fewer, it is kept whole; otherwise, a rolling window of 4 words is applied. We will refer to these smaller chunks as **ngrams** - mostly they will be 4 words long, but some may be shorter and some may be as long as 6 words.
```

Comment:
```suggestion
2. **Sentence chunking**: Then, if the sentence is 6 words or fewer, it is kept whole; otherwise, a rolling window of 4 words is applied. We will refer to these smaller chunks as **ngrams** - mostly they will be 4 words long, but some may be shorter and some may be as long as 6 words. We chose this rolling window size as it is similar to the lengths of the target phrases we match to in the next steps.
```